roc/log.go

```diff
@@ -181,6 +192,36 @@ func loggerRoutine() {
 	}
 }
 
+func getGID() uint64 {
+	b := make([]byte, 64)
+	b = b[:runtime.Stack(b, false)]
+	b = bytes.TrimPrefix(b, []byte("goroutine "))
+	b = b[:bytes.IndexByte(b, ' ')]
+	n, _ := strconv.ParseUint(string(b), 10, 64)
+	return n
+}
+
+func logWrite(level LogLevel, text string) {
+	if level >= LogDebug {
```

Comment:
We should check `level` against currently used log level, not just `LogDebug`. For that, we'll need to remember level in `SetLogLevel()`. To avoid data race, we can use atomic.

Reply:
Can you elaborate a bit more on this

Reply:
Your logErite func does tge following check: if given level is higher than debug, then print the message. 

Instead, the check should be: if given level is not higher than the level configured by user via SetLogLevel, then print it.

When the user invokes e.g. SetLogLevel(info), then only error and info messages should be printed, and debug and trace should be dropped.

See comment for SetLogLevel.